File: flutter/ios/Classes/Stream/ProgressUpdateHandler.swift
//
//  ProgressUpdateHandler.swift
//  spren-flutter
//
//  Created by nick on 22.03.2022.
//

import Foundation

class ProgressUpdateHandler: NSObject, FlutterStreamHandler {
    var eventSink: FlutterEventSink?
    
    func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
        self.eventSink = events
        return nil
    }
    
    func onCancel(withArguments arguments: Any?) -> FlutterError? {
        eventSink = nil
        return nil
    }
}

File: flutter/android/src/main/java/com/spren/spren_flutter/stream/ProgressUpdateHandler.java
package com.spren.spren_flutter.stream;

import io.flutter.plugin.common.EventChannel.EventSink;
import io.flutter.plugin.common.EventChannel.StreamHandler;

public class ProgressUpdateHandler implements StreamHandler {
    private static final ProgressUpdateHandler instance = new ProgressUpdateHandler();
    private EventSink eventSink;

    private ProgressUpdateHandler(){ }

    public static ProgressUpdateHandler getInstance() {
        return instance;
    }

    @Override
    public void onListen(Object arguments, EventSink eventSink) {
        this.eventSink = eventSink;
    }

    @Override
    public void onCancel(Object arguments) {
        this.eventSink = null;
    }

    public EventSink getEventSink() {
        return eventSink;
    }
}
